### PR TITLE
fix(express-engine): change provider assignable type from Provider to…

### PR DESCRIPTION
**Pull request by LennardWesterveld**
I'm submitting a ...
```
- [x] bug report
- [ ] feature request
```

```
- [x] The commit message follows our guidelines: https://github.com/angular/universal/blob/master/CONTRIBUTING.md#commit-message-format
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
```
Tested in combination of pull request of #774

* **What modules are related to this pull-request**
```
- [ ] aspnetcore-engine
- [x] express-engine
- [ ] hapi-engine
```

* **What is the current behavior?**
express-engine gives error while compiling the typescript for the server.

* **What is the new behavior (if this is a feature change)?**
Adds compatibility with the latest version of Angular (v5.0.0-beta.4)

* **Does this PR introduce a breaking change?** 
No

* **Please tell us about your environment:**
Angular version: 5.0.0-beta.3

### CODE REVIEW

**Comment by LennardWesterveld** [comment]
I signed it!

File: modules/ng-express-engine/src/main.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ import { REQUEST, RESPONSE } from './tokens';
  */
 export interface NgSetupOptions {
   bootstrap: Type<{}> | NgModuleFactory<{}>;
-  providers?: Provider[];
+  providers?: StaticProvider[];
```
Comment (Toxicable) [inline modules/ng-express-engine/src/main.ts]: Looks like a breaking change
Maybe `Provider[] | StaticProvider[]` might be better here

Comment (LennardWesterveld) [reply modules/ng-express-engine/src/main.ts]: That will not work i tried it out get this error 'Cannot invoke an expression whose type lacks a call signature. Type '{ (...items: StaticProvider[][]): StaticProvider[]; (...items: (any[] | ValueProvider | StaticCla...' has no compatible call signatures.'

**Comment by vikerman** [comment]
Thanks. Handled by https://github.com/angular/universal/commit/5c705cb51362bd448c0b8ac293da867a3a1913f4

**Pull request closed without merging**